tensorflow/core/kernels/sparse_xent_op_test.cc

```diff
@@ -41,40 +41,52 @@ static Graph* SparseXent(int batch_size, int num_classes) {
   return g;
 }
 
-#define BM_SparseXentDev(BATCH, CLASS, DEVICE)                          \
-  static void BM_SparseXent##_##BATCH##_##CLASS##_##DEVICE(int iters) { \
-    testing::ItemsProcessed(static_cast<int64>(iters) * BATCH * CLASS); \
-    test::Benchmark(#DEVICE, SparseXent(BATCH, CLASS)).Run(iters);      \
-  }                                                                     \
-  BENCHMARK(BM_SparseXent##_##BATCH##_##CLASS##_##DEVICE);
+#define BM_SparseXentDev(BATCH, CLASS, DEVICE, DTType)                     \
+  static void BM_SparseXent##_##BATCH##_##CLASS##_##DEVICE_##DTType(       \
+      int iters) {                                                         \
+    testing::ItemsProcessed(static_cast<int64>(iters) * BATCH * CLASS);    \
+    test::Benchmark(#DEVICE, SparseXent(BATCH, CLASS, DTType)).Run(iters); \
+  }                                                                        \
+  BENCHMARK(BM_SparseXent##_##BATCH##_##CLASS##_##DEVICE_##DTType);
 
 /// The representative tests for ptb_word on GPU
 #if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
-BM_SparseXentDev(8, 1000000, gpu);
+BM_SparseXentDev(8, 1000000, gpu, DT_FLOAT);
 
-BM_SparseXentDev(16, 10000, gpu);
-BM_SparseXentDev(16, 30000, gpu);
-BM_SparseXentDev(16, 100000, gpu);
+BM_SparseXentDev(16, 10000, gpu, DT_FLOAT);
+BM_SparseXentDev(16, 30000, gpu, DT_FLOAT);
+BM_SparseXentDev(16, 100000, gpu, DT_FLOAT);
 
-BM_SparseXentDev(32, 10000, gpu);
-BM_SparseXentDev(32, 30000, gpu);
-BM_SparseXentDev(32, 100000, gpu);
+BM_SparseXentDev(32, 10000, gpu, DT_FLOAT);
+BM_SparseXentDev(32, 30000, gpu, DT_FLOAT);
+BM_SparseXentDev(32, 100000, gpu, DT_FLOAT);
 
-BM_SparseXentDev(64, 10000, gpu);
-BM_SparseXentDev(64, 30000, gpu);
-BM_SparseXentDev(64, 100000, gpu);
+BM_SparseXentDev(64, 10000, gpu, DT_FLOAT);
+BM_SparseXentDev(64, 30000, gpu, DT_FLOAT);
+BM_SparseXentDev(64, 100000, gpu, DT_FLOAT);
 #endif  // GOOGLE_CUDA || TENSORFLOW_USE_ROCM
 
 // CPU
-BM_SparseXentDev(8, 1000000, cpu);
+BM_SparseXentDev(8, 1000000, cpu, DT_FLOAT);
 
-BM_SparseXentDev(16, 10000, cpu);
-BM_SparseXentDev(16, 100000, cpu);
+BM_SparseXentDev(16, 10000, cpu, DT_FLOAT);
+BM_SparseXentDev(16, 100000, cpu, DT_FLOAT);
 
-BM_SparseXentDev(32, 10000, cpu);
-BM_SparseXentDev(32, 100000, cpu);
+BM_SparseXentDev(32, 10000, cpu, DT_FLOAT);
+BM_SparseXentDev(32, 100000, cpu, DT_FLOAT);
 
-BM_SparseXentDev(64, 10000, cpu);
-BM_SparseXentDev(64, 100000, cpu);
+BM_SparseXentDev(64, 10000, cpu, DT_FLOAT);
+BM_SparseXentDev(64, 100000, cpu, DT_FLOAT);
+
+BM_SparseXentDev(8, 1000000, cpu, DT_BFLOAT16);
+
+BM_SparseXentDev(16, 10000, cpu, DT_BFLOAT16);
+BM_SparseXentDev(16, 100000, cpu, DT_BFLOAT16);
+
+BM_SparseXentDev(32, 10000, cpu, DT_BFLOAT16);
+BM_SparseXentDev(32, 100000, cpu, DT_BFLOAT16);
+
+BM_SparseXentDev(64, 10000, cpu, DT_BFLOAT16);
+BM_SparseXentDev(64, 100000, cpu, DT_BFLOAT16);
```

Comment:
Nit: Since the cases for both float and bfloat16 are the same, could you please define another macro that takes a dtype and calls `BM_SparseXentDev` for all these sizes on CPU? For example,
```c++
#define BM_SPARSE_XENT_DEV_CPU(DTYPE).      \
  BM_SparseXentDev(8, 1000000, cpu, DTYPE); \
  BM_SparseXentDev(16, 10000, cpu, DTYPE);  \
  BM_SparseXentDev(16, 100000, cpu, DTYPE); \
  BM_SparseXentDev(32, 10000, cpu, DTYPE);  \
  BM_SparseXentDev(32, 100000, cpu, DTYPE); \
  BM_SparseXentDev(64, 10000, cpu, DTYPE);  \
  BM_SparseXentDev(64, 100000, cpu, DTYPE);

BM_SPARSE_XENT_DEV_CPU(DT_FLOAT);
BM_SPARSE_XENT_DEV_CPU(DT_BFLOAT16);
```